### PR TITLE
Derive workspace name from namespace labels

### PR DESCRIPTION
**Pull request by wozniakjan**
## Issue ID(s):
The new style workspaces have different way of figuring out a namespace. This PR should ensure both old style orgs as well as new style workspaces work.

### CODE REVIEW

File: pkg/bot.go [hunk 1]
```diff
@@ -497,23 +500,38 @@ func (b *bot) deletePreviewSite(args ResponseRequest, verboseErrors bool) string
...
 
-func (b *bot) hasPreviewSiteCapability(email, org string) (bool, error) {
+func (k kubeClients) getWorkspace(org string) string {
+	ns, err := k.nsLister.Get(org)
+	if err != nil {
+		klog.Errorf("failed to find namespace %v: %v", org, err)
+		return org
+	}
+	// for legacy organizations, ws == namespace == org
+	if ns.Labels["ddev.live/displayname"] != "" && ns.Labels["ddev.live/subscriptionstub"] != "" {
```
Comment (drekle) [inline pkg/bot.go:510]: I need to change one of these constants prior to consumption however:
https://github.com/drud/api-common/blob/main/utils/workspace.go#L10

for later...

**Review by drekle — APPROVED** [review]
lgtm

File: pkg/bot.go [hunk 2]
```diff
@@ -497,23 +500,38 @@ func (b *bot) deletePreviewSite(args ResponseRequest, verboseErrors bool) string
...
+	return org
+}
+
+func (k kubeClients) hasPreviewSiteCapability(email, org string) (bool, string, error) {
 	ctx := context.Background()
-	metaCtx := grpcmeta.AppendToOutgoingContext(ctx, "x-ddev-workspace", org)
-	caps, err := b.authClient.ListCapabilities(metaCtx, &pbAdmin.ListCapabilitiesRequest{Email: email})
+	ws := k.getWorkspace(org)
+	metaCtx := grpcmeta.AppendToOutgoingContext(ctx, "x-ddev-workspace", ws)
+	caps, err := k.authClient.ListCapabilities(metaCtx, &pbAdmin.ListCapabilitiesRequest{Email: email})
```
Comment (nic-hima) [inline pkg/bot.go:521]: good check here

**Review by nic-hima — APPROVED** [review]
looks to me like this could handle the new workspace changes

**Pull request merged**